File: lib/models/conversation.model.ts
import mongoose from "mongoose";
// Définition du schéma d'un message
const messageSchema = new mongoose.Schema({
  sender: {
    type: String,
    required: true,
  },
  text: {
    type: String,
    required: true,
  },
  createdAt: {
    type: Date,
    default: Date.now,
  },
});
// Définition du schéma d'une conversation
const conversationSchema = new mongoose.Schema({
  name: {
    type: String,
    required: true,
    default: "New Conversation",
  },
  participants: [
    {
      type: String,
      ref: "User",
      required: true,
    },
  ],
  messages: [
    {
      type: String,
      ref: "messages",
      required: true,
    },
  ],
  createdAt: {
    type: Date,
    default: Date.now,
  },
  updatedAt: {
    type: Date,
    default: Date.now,
  },
});

const Conversation =
  mongoose.models.Conversation ||
  mongoose.model("Conversation", conversationSchema);

export default Conversation;

File: components/chatapp/models/conversationmodel.py
import datetime
from mongoengine import Document, StringField, ListField, DateTimeField, ReferenceField
from .messagemodel import Message

class Conversation(Document):
    name = StringField(required=True, default='New Conversation')
    participants = ListField(StringField(), required=True)
    messages = ListField(ReferenceField(Message))
    createdAt = DateTimeField(default=datetime.datetime.now)
    updatedAt = DateTimeField(default=datetime.datetime.now)

    meta = {
        'collection': 'conversations'  # Optional: Specify the collection name
    }